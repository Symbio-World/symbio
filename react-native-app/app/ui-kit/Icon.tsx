import * as React from 'react'

import { SvgProps } from 'react-native-svg'
import camera from '../../assets/svg/camera.svg'
import cross from '../../assets/svg/cross.svg'
import disk from '../../assets/svg/disk.svg'
import flashlight from '../../assets/svg/flashlight.svg'
import historyicon from '../../assets/svg/history.svg'
import fatalError from '../../assets/svg/fogg-fatal-error.svg'

type Props = SvgProps & {
  icon: Icons
}

export enum Icons {
  CAMERA = 'CAMERA',
  CROSS = 'CROSS',
  DISK = 'DISK',
  FLASHLIGHT = 'FLASHLIGHT',
  HISTORYICON = 'HISTORYICON',
  FATAL_ERROR = 'FATAL_ERROR',
}

const IconsMap = {
  [Icons.CAMERA]: camera,
  [Icons.CROSS]: cross,
  [Icons.DISK]: disk,
  [Icons.FLASHLIGHT]: flashlight,
  [Icons.HISTORYICON]: historyicon,
  [Icons.FATAL_ERROR]: fatalError,
}

export const Icon: React.FC<Props> = ({ icon, ...otherProps }) => {
  const IconComponent = IconsMap[icon]
  if (!IconComponent) throw new Error('No icon found')
  return <IconComponent {...otherProps} />
}
