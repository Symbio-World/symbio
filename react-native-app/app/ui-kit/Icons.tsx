import * as React from 'react'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

export 

