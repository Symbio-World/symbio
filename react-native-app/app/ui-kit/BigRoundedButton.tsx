import * as React from 'react'
import { TouchableOpacity, Text, View } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, Icons } from './Icon'

type Props = {
  title: string
  icon: React.ReactNode
  onPress?: () => void
  variant?: 'outline' | 'fill'
}

const ButtonStyle = {
  fill: [t.bgPrimary],
  outline: [t.bgWhite, t.border2, t.borderPrimary],
}

const TitleStyle = {
  fill: [t.textWhite],
  outline: [t.textPrimary],
}

export const BigRoundedButton: React.FC<Props> = ({
  title,
  icon,
  onPress = () => {},
  variant = 'fill',
}) => {
  const buttonStyle = ButtonStyle[variant]
  const titleStyle = TitleStyle[variant]

  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={[
          ...buttonStyle,
          t.w24,
          t.h24,
          t.roundedFull,
          t.flexCol,
          t.itemsCenter,
          t.justifyCenter,
        ]}
      >
        <View style={[t.w6, t.h6, t.mB2]}>{icon}</View>
        <View>
          <Text style={[t.textCenter, t.textBase, t.fontBold, ...titleStyle]}>
            {title}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  )
}
