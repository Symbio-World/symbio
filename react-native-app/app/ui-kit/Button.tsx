import * as React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, IconsEnum } from './Icon'

export enum ButtonStyle {
  FILL,
  OUTLINE,
  TRANSPARENT,
}

type Props = {
  onPress?: () => void
  title?: string
  icon?: IconsEnum
  variant?: ButtonStyle
}

const ButtonBaseStyle = {
  [ButtonStyle.FILL]: [t.bgPrimary],
  [ButtonStyle.OUTLINE]: [t.bgWhite, t.border2, t.borderPrimary],
  [ButtonStyle.TRANSPARENT]: [t.bgTransparent],
}

const InnerTitleBaseStyle = {
  [ButtonStyle.FILL]: [t.textWhite],
  [ButtonStyle.OUTLINE]: [t.textPrimary],
  [ButtonStyle.TRANSPARENT]: [t.textPrimary],
}

export const Button: React.FC<Props> = ({
  onPress = () => {},
  title = '',
  variant = ButtonStyle.FILL,
  icon = null,
}) => {
  const buttonStyle = ButtonBaseStyle[variant]
  const innerStyle = [InnerTitleBaseStyle[variant]]
  return (
    <TouchableOpacity
      style={[
        t.p4,
        t.itemsCenter,
        t.justifyCenter,
        t.flex,
        t.flexRow,
        t.rounded,
        ...buttonStyle,
      ]}
      onPress={onPress}
    >
      {icon ? <Icon style={[t.mR2]} iconType={icon} /> : null}
      <Text style={[t.textCenter, t.text2xl, t.textWhite, ...innerStyle]}>
        {title}
      </Text>
    </TouchableOpacity>
  )
}
