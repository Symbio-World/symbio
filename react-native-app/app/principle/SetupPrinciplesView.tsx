import * as React from 'react'
import { ScrollView, View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { PrincipleView } from './PrincipleView'
import { Button } from '../ui-kit/Button'

type Props = {
  principles?: string[]
  onSubmit?: (selectedPrinciples: string[]) => void
}
export const SetupPrinciplesView: React.FC<Props> = ({
  principles = [],
  onSubmit = () => {},
}) => {
  const [selectedPrinciples, setSelectedPrinciples] = React.useState<string[]>(
    [],
  )

  const onPrinciplePress = (principle: string) => {
    if (selectedPrinciples.includes(principle)) {
      setSelectedPrinciples(selectedPrinciples.filter((p) => p !== principle))
    } else {
      setSelectedPrinciples([...selectedPrinciples, principle])
    }
  }

  const handleSubmit = () => {
    onSubmit(selectedPrinciples)
  }

  return (
    <View style={[t.m5]}>
      <ScrollView>
        <View>
          <Text style={[t.text4xl, t.fontBold, t.mY4]}>ðŸ‘‹Hey There</Text>
          <Text style={[t.mY4]}>
            <Text style={[t.fontBold]}>Should I Buy</Text> lets you understand
            if the products you find in the supermarket fit your values.
          </Text>
          <Text style={[t.fontBold]}>What is important to you?</Text>
          <View style={[t.flexRow, t.flexWrap, t.mY4]}>
            {principles.map((p) => (
              <PrincipleView
                key={p}
                title={p}
                onPress={() => onPrinciplePress(p)}
                selected={selectedPrinciples.includes(p)}
              />
            ))}
          </View>
        </View>
        <View style={[t.h24]} />
      </ScrollView>
      <View style={[t.absolute, t.bottom0, t.wFull, t.bgWhite]}>
        <Button onPress={handleSubmit} title="Get Started" />
      </View>
    </View>
  )
}
