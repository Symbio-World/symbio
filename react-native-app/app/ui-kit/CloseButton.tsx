import * as React from 'react'

import { RoundedButton } from './RoundedButton'
import { ButtonStyle } from './Button'
import { Icons } from './Icon'

type CloseButtonProps = {
  onClose?: () => void
}

export const CloseButton: React.FC<CloseButtonProps> = ({ onClose }) => (
  <RoundedButton
    onPress={onClose}
    outerTitle="close"
    variant={ButtonStyle.OUTLINE}
    icon={Icons.CROSS}
  />
)
