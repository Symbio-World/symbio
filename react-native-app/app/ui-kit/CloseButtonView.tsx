import * as React from 'react'
import { t } from 'react-native-tailwindcss'
import { CloseButton } from './CloseButton'
import { View } from 'react-native'

type CloseButtonViewProps = {
  onPress?: () => void
}

export const CloseButtonView: React.FC<CloseButtonViewProps> = ({
  onPress,
}) => (
  <>
    <View style={[t.h40]} />
    <View style={[t.absolute, t.bottom0, t.wFull, t.itemsCenter, t.pB5]}>
      <CloseButton onPress={onPress} />
    </View>
  </>
)
