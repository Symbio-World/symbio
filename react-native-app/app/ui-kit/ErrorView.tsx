import * as React from 'react'
import { View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { Icon, Icons } from './Icon'
import { CloseButtonView } from './CloseButtonView'

type Props = {
  error?: unknown
  onPress?: () => void
}
export const ErrorView: React.FC<Props> = ({ onPress = () => {} }) => {
  return (
    <View style={[t.flex1]}>
      <Icon icon={Icons.FATAL_ERROR} />
      <Text style={[t.textCenter, t.textBlack, t.text5xl, t.mB10]}>
        OOOOPS!
      </Text>
      <Text style={[t.textCenter, t.textBlack, t.text3xl, t.mX12]}>
        Something went wrong... You can tell us what happened or try again later
      </Text>
      <CloseButtonView onPress={onPress} />
    </View>
  )
}
