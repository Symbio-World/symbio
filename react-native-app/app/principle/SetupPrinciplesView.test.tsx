import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import { Button } from '../ui-kit/Button'
import { SetupPrinciplesView } from './SetupPrinciplesView'
import { PrincipleView } from './PrincipleView'

describe('SetupPrinciplesView', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<SetupPrinciplesView />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders principles', () => {
    const principles = ['principle1', 'principle2', 'principle3']
    const { getAllByType } = render(
      <SetupPrinciplesView principles={principles} />,
    )
    expect(getAllByType(PrincipleView)).toHaveLength(principles.length)
  })

  it('calls onSubmit with selected principles', () => {
    const handleSubmit = jest.fn()
    const principles = ['principle1', 'principle2', 'principle3']
    const { getByType, getAllByType } = render(
      <SetupPrinciplesView principles={principles} onSubmit={handleSubmit} />,
    )
    const principleViews = getAllByType(PrincipleView)
    fireEvent(principleViews[0], 'onPress', principles[0])
    fireEvent(principleViews[1], 'onPress', principles[1])
    fireEvent.press(getByType(Button))
    expect(handleSubmit).toHaveBeenCalledWith([principles[0], principles[1]])
  })

  it('deselects principles', () => {
    const handleSubmit = jest.fn()
    const principles = ['principle1', 'principle2', 'principle3']
    const { getByType, getAllByType } = render(
      <SetupPrinciplesView principles={principles} onSubmit={handleSubmit} />,
    )
    const principleViews = getAllByType(PrincipleView)
    fireEvent(principleViews[0], 'onPress', principles[0])
    fireEvent(principleViews[0], 'onPress', principles[0])
    fireEvent.press(getByType(Button))
    expect(handleSubmit).toHaveBeenCalledWith([])
  })
})
