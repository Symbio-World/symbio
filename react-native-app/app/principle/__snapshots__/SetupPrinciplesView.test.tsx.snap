// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupPrinciplesView renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "margin": 20,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 36,
              },
              Object {
                "fontWeight": "bold",
              },
              Object {
                "marginVertical": 16,
              },
            ]
          }
        >
          ðŸ‘‹Hey There
        </Text>
        <Text
          style={
            Array [
              Object {
                "marginVertical": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Should I Buy
          </Text>
           lets you understand if the products you find in the supermarket fit your values.
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          What is important to you?
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "flexWrap": "wrap",
              },
              Object {
                "marginVertical": 16,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 96,
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "bottom": 0,
        },
        Object {
          "width": "100%",
        },
        Object {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#B35340",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#fff",
            },
            Array [
              Object {
                "color": "#fff",
              },
            ],
          ]
        }
      >
        Get Started
      </Text>
    </View>
  </View>
</View>
`;
